### PR TITLE
Editor: grunion_media_button not found fix

### LINKED ISSUE #544: Editor: grunion_media_button not found
This is a follow up to #528.

A user has reported that after the SiteOrigin Widgets Bundle 1.9.8 update, they now receive the following warning message when trying to edit a SiteOrigin Editor widget:

`Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'grunion_media_button' not found or invalid function name in /home/fondatioua/www/wp-includes/class-wp-hook.php on line 298`

They've stated they only experience this when TinyMCE advanced is enabled but I can't personally replicate that - it could very well be as a result of their toolbar settings. I'm logging this as we should be able to resolve this by simply checking if the grunion_media_button function exists before trying to remove it.

### PR DESCRIPTION
**Pull request by AlexGStapleton**
Resolves #544

### CODE REVIEW

File: base/inc/fields/tinymce.class.php [hunk 1]
```diff
@@ -517,14 +517,18 @@ private function render_media_buttons( $editor_id ) {
...
 		$screen = function_exists( 'get_current_screen' ) ? get_current_screen() : null;
-		// Temporarily disable the Jetpack Grunion contact form editor on the widgets screen.
-		if( ! is_null( $screen ) && $screen->id == 'widgets' ) {
-			remove_action( 'media_buttons', 'grunion_media_button', 999 );
-		}
-		do_action( 'media_buttons', $editor_id );
-		// Temporarily disable the Jetpack Grunion contact form editor on the widgets screen.
-		if( ! is_null( $screen ) && $screen->id == 'widgets' ) {
-			add_action( 'media_buttons', 'grunion_media_button', 999 );
+		if ( function_exists( 'grunion_media_button' ) && ! is_null( $screen ) ) {
```
Comment (braamgenis) [inline base/inc/fields/tinymce.class.php]: Can keep this as only 2 `if` statements and make it all a bit cleaner by assigning the boolean expression to a variable i.e.
```
        // Temporarily disable the Jetpack Grunion contact form editor on the widgets screen.
		$disable_grunion = function_exists( 'grunion_media_button' ) && ! is_null( $screen ) && $screen->id == 'widgets';
		if( $disable_grunion ) {
			remove_action( 'media_buttons', 'grunion_media_button', 999 );
		}
		do_action( 'media_buttons', $editor_id );
		// Temporarily disable the Jetpack Grunion contact form editor on the widgets screen.
		if( $disable_grunion ) {
			add_action( 'media_buttons', 'grunion_media_button', 999 );
		}
```

**Comment by AlexGStapleton** [comment]
Hm. Is this even required anymore? I ask as it appears as [they now only output this code purely for the posts page](https://github.com/Automattic/jetpack/blob/master/modules/contact-form/grunion-editor-view.php#L18).

**Comment by AlexGStapleton** [comment]
Well regardless, just in case it is I've committed the desired adjustments.

**Comment by braamgenis** [comment]
Yeah, looks like it might not be necessary anymore. Can give it a test and see if the issue is fixed without this PR?

**Comment by AlexGStapleton** [comment]
Closing in favor of #643

**Pull request closed without merging**